index.d.ts

```diff
@@ -0,0 +1,16 @@
+declare module 'ember' {
+  /**
+   * NOTE: because of the way @types/ember works,
+   * merging a few properties into the Ember.Route interface
+   * will also result in the @ember/routing/route default export
+   * having these properties too.
+   * -- @mike-north
+   */
+  namespace Ember {
+    interface Route {
+      titleToken: string;
+      title: string;
```

Comment:
Is this right? Both `titleToken` and `title` can be either a string or a function that can return a few different things.

Reply:
The type information does need to change to accommodate this. Does the function always return a string?

Reply:
`title` can be

- A string
- A function that returns a string

`titleToken` can be

- A string
- A function that returns
  - A string
  - An array of strings
  - And new in the yet to be released version:
    - A promise that resolves to a string
    - An array of promises that resolves to a string